diff --git a/lib/src/enums.dart b/lib/src/enums.dart
index 9f134e3..9024bc6 100644
--- a/lib/src/enums.dart
+++ b/lib/src/enums.dart
@@ -176,3 +176,31 @@ RTCPeerConnectionState peerConnectionStateForString(String? state) {
 
   return RTCPeerConnectionState.RTCPeerConnectionStateClosed;
 }
+
+enum RTCDegradationPreference {
+  DISABLED,
+  MAINTAIN_FRAMERATE,
+  MAINTAIN_RESOLUTION,
+  BALANCED,
+}
+
+final typeRTCDegradationPreferenceString = <RTCDegradationPreference, String>{
+  RTCDegradationPreference.DISABLED: 'disabled',
+  RTCDegradationPreference.MAINTAIN_FRAMERATE: 'maintain-framerate',
+  RTCDegradationPreference.MAINTAIN_RESOLUTION: 'maintain-resolution',
+  RTCDegradationPreference.BALANCED: 'balanced',
+};
+
+RTCDegradationPreference degradationPreferenceforString(String? degradation) {
+  switch (degradation) {
+    case 'disabled':
+      return RTCDegradationPreference.DISABLED;
+    case 'maintain-framerate':
+      return RTCDegradationPreference.MAINTAIN_FRAMERATE;
+    case 'maintain-resolution':
+      return RTCDegradationPreference.MAINTAIN_RESOLUTION;
+    case 'balanced':
+      return RTCDegradationPreference.BALANCED;
+  }
+  return RTCDegradationPreference.BALANCED;
+}
diff --git a/lib/src/rtc_rtp_parameters.dart b/lib/src/rtc_rtp_parameters.dart
index 41d3300..20c5577 100644
--- a/lib/src/rtc_rtp_parameters.dart
+++ b/lib/src/rtc_rtp_parameters.dart
@@ -1,3 +1,4 @@
+import 'enums.dart';
 import 'rtc_rtcp_parameters.dart';
 
 class RTCRTPCodec {
@@ -151,6 +152,7 @@ class RTCRtpParameters {
     this.headerExtensions,
     this.encodings,
     this.codecs,
+    this.degradationPreference,
   });
 
   factory RTCRtpParameters.fromMap(Map<dynamic, dynamic> map) {
@@ -169,12 +171,16 @@ class RTCRtpParameters {
     codecsMap.forEach((params) {
       codecs.add(RTCRTPCodec.fromMap(params));
     });
+
+    var degradationPreference = map['degradationPreference'];
     var rtcp = RTCRTCPParameters.fromMap(map['rtcp']);
     return RTCRtpParameters(
         transactionId: map['transactionId'],
         rtcp: rtcp,
         headerExtensions: headerExtensions,
         encodings: encodings,
+        degradationPreference:
+            degradationPreferenceforString(degradationPreference),
         codecs: codecs);
   }
 
@@ -186,6 +192,8 @@ class RTCRtpParameters {
 
   List<RTCRtpEncoding>? encodings;
 
+  RTCDegradationPreference? degradationPreference;
+
   /// Codec parameters can't currently be changed between getParameters and
   /// setParameters. Though in the future it will be possible to reorder them or
   /// remove them.
@@ -210,6 +218,9 @@ class RTCRtpParameters {
       'headerExtensions': headerExtensionsList,
       'encodings': encodingList,
       'codecs': codecsList,
+      if (degradationPreference != null)
+        'degradationPreference':
+            typeRTCDegradationPreferenceString[degradationPreference!],
     };
   }
 }
